oci/private/authn.bzl

```diff
@@ -2,53 +2,42 @@
 
 load("@aspect_bazel_lib//lib:base64.bzl", "base64")
 load("@aspect_bazel_lib//lib:repo_utils.bzl", "repo_utils")
+load("//oci/private:toolchains_repo.bzl", "PLATFORMS")
 load(":util.bzl", "util")
 
-
-# Unfortunately bazel downloader doesn't let us sniff the WWW-Authenticate header, therefore we need to
-# keep a map of known registries that require us to acquire a temporary token for authentication.
-_WWW_AUTH = {
-    "index.docker.io": {
-        "realm": "auth.docker.io/token",
-        "scope": "repository:{repository}:pull",
-        "service": "registry.docker.io",
-    },
-    "public.ecr.aws": {
-        "realm": "{registry}/token",
-        "scope": "repository:{repository}:pull",
-        "service": "{registry}",
-    },
-    "ghcr.io": {
-        "realm": "{registry}/token",
-        "scope": "repository:{repository}:pull",
-        "service": "{registry}/token",
-    },
-    "cgr.dev": {
-        "realm": "{registry}/token",
-        "scope": "repository:{repository}:pull",
-        "service": "{registry}",
-    },
-    ".azurecr.io": {
-        "realm": "{registry}/oauth2/token",
-        "scope": "repository:{repository}:pull",
-        "service": "{registry}",
-    },
-    "registry.gitlab.com": {
-        "realm": "gitlab.com/jwt/auth",
-        "scope": "repository:{repository}:pull",
-        "service": "container_registry",
-    },
-    ".app.snowflake.com": {
-        "realm": "{registry}/v2/token",
-        "scope": "repository:{repository}:pull",
-        "service": "{registry}",
-    },
-    "docker.elastic.co": {
-        "realm": "docker-auth.elastic.co/auth",
-        "scope": "repository:{repository}:pull",
-        "service": "token-service",
-    },
-}
+_default_www_auth = [
```

Comment:
We could actually remove the whitelist and have all requests go through the `crane auth token` logic. Can I do that?